.evergreen/config.yml

```diff
@@ -664,29 +635,33 @@ functions:
         working_dir: "src"
         shell: "bash"
         script: |
-          ${PREPARE_SHELL}
           if [ "${skip_EC2_auth_test}" = "true" ]; then
              echo "This platform does not support the EC2 auth test, skipping..."
              exit 0
           fi
-          set -ex
+
+          # Disable xtrace for security reasons (just in case it was accidentally set).
+          set +x
+          set -o errexit
+
+          # Try to source exported AWS Secrets
+          if [ -f ./secrets-export.sh ]; then
+            source ./secrets-export.sh
+          fi
+
           cd ${DRIVERS_TOOLS}/.evergreen/auth_aws
+
           . ./activate-authawsvenv.sh
+
           python aws_tester.py ec2
-    - command: shell.exec
-      type: test
-      params:
-        working_dir: "src"
-        script: |
-          ${PREPARE_SHELL}
-          if [ "${skip_EC2_auth_test}" = "true" ]; then
-             echo "This platform does not support the EC2 auth test, skipping..."
-             exit 0
-          fi
-          # Write an empty prepare_mongodb_aws so no auth environment variables
-          # are set.
-          rm "${PROJECT_DIRECTORY}/prepare_mongodb_aws.sh" || true
-          PYTHON_BINARY=${PYTHON_BINARY} ASSERT_NO_URI_CREDS=true .evergreen/run-mongodb-aws-test.sh
+
+          cd -
+
+          # Write an empty prepare_aws_env so no auth environment variables are set.
```

Comment:
We shouldn't need to do this anymore (perhaps something still needs to change in D-E-T?

Reply:
Ah, I see how it is used now.  I think we can also absorb this logic into `run-mongodb-aws-test.sh`.  I do like the idea of adding the script in D-E-T as you've done.